components/brave_federated/config_utils.cc

```diff
@@ -0,0 +1,129 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated/config_utils.h"
+
+#include "base/json/json_reader.h"
+#include "brave/components/brave_federated/api/config.h"
+#include "brave/components/brave_federated/features.h"
+
+namespace brave_federated {
+
+LearningServiceConfig::LearningServiceConfig() {
+  reconnect_policy_ = {.num_errors_to_ignore = 0,
+                       .initial_delay_ms = 10 * 1000,
+                       .multiply_factor = 2.0,
+                       .jitter_factor = 0.0,
+                       .maximum_backoff_ms = 60 * 10 * 1000,
```

Comment:
bug prone multiplication

Reply:
is this because it's a multiplication in the initialiser? Should we just define these as constants in an anonymous namespace?

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/bugprone/implicit-widening-of-multiplication-result.html

Reply:
nit: rather than magic numbers we could use base::Time constants?

Reply:
switched to base::Time constants. I'm not seeing the issue with the multiplication, the linked article says this is only a problem when the product may overflow